### ISSUE TITLE
State recovering after HA start

### ISSUE THREAD

**Issue by andrewjswan** [custom-components/ble_monitor#176]
Recovering the state after the start is now working for all types of sensors, whether it is possible to make a separate option to recover the values of the battery state, the state of conventional sensors and the state of binary sensors (I am not sure that this is necessary at all). 
Or maybe a separate option per device (Battery recovery status, I think you can make it global because adding a given sensor is global).

**Comment by andrewjswan** [comment]
@Magalex2x14 And by chance there is no way to query the state of binary sensors after the start?  After all, MiHome sees the state somehow.

**Comment by Magalex2x14** [comment]
MiHome is not designed to monitor the status of sensors 24/7, MiHome simply connects to a sensor and queries its status as far as I understand. In addition, the process of connecting with such a sensor is also not so simple - the sensor must be open (the door is open), in any case, it is only available to me in the application in this way. If there was an opportunity to provoke a sensor to send messages about its state, I would definitely think about its implementation. Unfortunately, the only way I know is to remove and insert its battery.
Perhaps this will be fixed in the next firmware update of the device, but for now, we have to live with it.

**Comment by andrewjswan** [comment]
I receive notifications from MiHome while at work when I open the door at
home. I also see the status of the door and lighting.

чт, 10 дек. 2020 г., 15:32 Aleksey Makarenko <notifications@github.com>:

**Comment by Magalex2x14** [comment]
This is the job of the gateway, not the MiHome application itself.

**Comment by Magalex2x14** [comment]
In addition, you can receive notification of the **opening event** from HA, if the ble_monitor was running at the time, and it does not depend on the previous state of the sensor.
Another question is what happens if the gateway is turned off, the door is opened, and the gateway is turned on?

**Comment by andrewjswan** [comment]
> Another question is what happens if the Xiaomi gateway is turned off, the door is opened, and the gateway is turned on?

Need test ...

**Comment by andrewjswan** [comment]
> Another question is what happens if the Xiaomi gateway is turned off, the door is opened, and the gateway is turned on?

Turn of gateway, open door, Turn on gateway, open MiHome, open sensor, state Door opened ...

I try check another:
- Open MiHome (BT turned off)
- Turn of gateway 
- Open door
- Turn on gateway

**Comment by Ernst79** [comment]
A per device setting for `restore_state` will be added with #213

**Comment by andrewjswan** [comment]
> A per device setting for `restore_state` will be added with

How about yaml file?

**Comment by Ernst79** [comment]
Should work with YAML as well. Docs haven’t been prepared yet, but the format will be like this.

```
ble_monitor:
  restore_state: False
  devices:
    - mac: 'A4:C1:38:2F:86:6C'
      restore_state: True
```

**Comment by Ernst79** [comment]
`restore_state` can now be overruled for individual devices. It's included in version 1.0.0, which has just been released.
